### PR TITLE
add Platon recipe

### PR DESCRIPTION
**Pull request by oschwengers**
Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the `please review & merge` label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it. Note that if you are not already a member of the bioconda project (meaning that you can't add this label), please ping `@bioconda/core` so that your PR can be reviewed and merged (please note if you'd like to be added to the bioconda project). Please see https://github.com/bioconda/bioconda-recipes/issues/15332 for more details.

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by epruesse** [comment]
The package downloads a file "https://s3.computational.bio.uni-giessen.de/swift/v1/platon/db.tar.gz" which doesn't seem to be available any more.

**Comment by oschwengers** [comment]
@epruesse : The URL is correct but we're just in the middle of a scheduled maintenance outage right now. This is a long planned update of our complete compute and cloud/S3 infrastructure which might last a couple of days. As soon as everything is back online, I'll trigger circleci again.

**Comment by epruesse** [comment]
@oschwengers You must not  download ***anything*** in the `build.sh`. Consider that 15 years from now, this package may have to be rebuilt. We keep copies of the sources listed in `source/url`, but we have *no* way of parsing build scripts for this kind of thing.

***IFF** the database is versioned synchronously with the source code and small, you could add it as another URL. It would be preferable for it to be released as part of the source archive, however.

If the database is large, it should be hosted somewhere else and downloaded by the user.

**Comment by oschwengers** [comment]
@BiocondaBot rebuild

**Comment by oschwengers** [comment]
```
$ hash & extract packages for noarch: 0% 0/1 [00:00<?, ?it/s]Too long with no output (exceeded 10m0s)
```
Might this be related to the package size? Is there a certain size limit for a conda package?
For the sake of user convenience I included a necessary DB (2.8 Gb) into the package. Is it ok or do I need to exclude the database?

**Comment by oschwengers** [comment]
@BiocondaBot rebuild

**Comment by oschwengers** [comment]
@epruesse : Thanks for the hint. I'll exclude the db download and ask users to manually download and setup the db.

File: recipes/platon/build.sh [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#!/bin/sh
+set -e
+
+# install PLATON via pip
+$PYTHON -m pip install --no-deps --ignore-installed --no-cache-dir .
```
Comment (bgruening) [inline recipes/platon/build.sh]: This could be moved directly into the meta.yml as build/script if you like.

File: recipes/platon/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  noarch: python
+  number: 0
+  entry_points:
+    - platon = platon.platon:main
+
+requirements:
+  host:
+    - python >=3
+
+  build:
```
Comment (bgruening) [inline recipes/platon/meta.yaml]: build should be moved up and host should be the second section.
Also pip should be in host not in build.

tar is not needed anymore?

File: recipes/platon/build.sh (hunk 1 above)
Comment (oschwengers) [reply recipes/platon/build.sh]: Ok. Good point, now that the necessary database download is passed to the user's responsibility.

File: recipes/platon/meta.yaml (hunk 2 above)
Comment (oschwengers) [reply recipes/platon/meta.yaml]: Ok, I wasn't aware of this. 

Tar is still needed as it's the source of the local pip installation. This way, it merely depends on the released GitHub version but not on a published PyPi.org-release.

**Comment by oschwengers** [comment]
Somehow the `macos` test is blocked. `Lint` + `linux` have successfully passed.
Any ideas?

**Comment by oschwengers** [comment]
@BiocondaBot rebuild

**Comment by bgruening** [comment]
If you are using wget in your post-link step you also need to add it to your run environment. You can test this locally with `conda build ./recipes/platon/`

**Pull request merged**